syntax = "proto3";

service Election {
  rpc Elect(ElectionRequest) returns (ElectionResponse);
  rpc AnnounceLeader(LeaderRequest) returns (LeaderResponse);
}

message ElectionRequest {
  string node_ip = 1;
}

message ElectionResponse {
  bool ok = 1;
}

message LeaderRequest {
  string leader_ip = 1;
}

message LeaderResponse {
  bool ok = 1;
}
